crates/core_simd/src/vector.rs

```diff
@@ -3,48 +3,66 @@ use crate::simd::{
     SimdPartialOrd, SupportedLaneCount, Swizzle,
 };
 
-/// A SIMD vector of `LANES` elements of type `T`. `Simd<T, N>` has the same shape as [`[T; N]`](array), but operates like `T`.
+/// A SIMD vector with the shape of `[T; N]` but the operations of `T`.
 ///
-/// Two vectors of the same type and length will, by convention, support the operators (+, *, etc.) that `T` does.
-/// These take the lanes at each index on the left-hand side and right-hand side, perform the operation,
-/// and return the result in the same lane in a vector of equal size. For a given operator, this is equivalent to zipping
-/// the two arrays together and mapping the operator over each lane.
+/// `Simd<T, N>` supports the operators (+, *, etc.) that `T` does in "elementwise" fashion.
+/// These take the element at each index from the left-hand side and right-hand side,
+/// perform the operation, then return the result in the same index in a vector of equal size.
+/// In other words, an elementwise operation is equivalent to a zip, then map.
 ///
```

Comment:
I'd add a single-line paragraph after this paragraph but before the code example that's something like

> The major difference between `Simd<T, N>` and `[T; N]` is that SIMD data generally has a higher required alignment.

That helps quickly answer the common question of "why are we making a separate type at all? why can't we just add this directly on arrays?"

Reply:
may also want to mention that arrays and vectors are different types in most compilers (e.g. LLVM IR) and using array types often won't produce simd instructions.

Reply:
"often won't" might overstate it  -- we have tests confirming that they often *will*

https://github.com/rust-lang/rust/blob/a73288371e3fa0a610fbc11e7e8418017bdfde42/tests/codegen/array-map.rs#L9-L27

Reply:
well, i built a more complex test (simplified core of a 3D triangle rasterizer) and i'm rather impressed by how good llvm has gotten since I tried a similar algorithm with clang and llvm 4:
https://rust.godbolt.org/z/dqq164sbx

Reply:
though it still needed some adjustment to make it vectorize (llvm won't convert a loop into a masked store), hence i think arrays still sometimes don't optimize as well as explicit simd ops.

Reply:
Certainly, but that's not because of the *type*, that's because of the operations.  If one doesn't use a masked load operation with a SIMD type it's probably not going to optimize to one either.